lib/model/classycat_classify.py

```diff
@@ -0,0 +1,129 @@
+import os
+import json
+import uuid
+import httpx
+from openai import OpenAI
+from lib.logger import logger
+from lib.model.model import Model
+from lib.schemas import Message, ClassyCatBatchClassificationResponse
+from lib.s3 import load_file_from_s3, file_exists_in_s3, upload_file_to_s3
+
+
+class LLMClient:
+    def __init__(self):
+        self.client = None
+
+    def get_client(self):
+        pass
+
+    def classify(self, task_prompt, items_count, max_tokens_per_item=50):
+        pass
+
+
+class OpenRouterClient(LLMClient):
+    def __init__(self, model_name):
+        super().__init__()
+        self.model_name = model_name
+
+    def get_client(self):
+        if self.client is None:
+            self.client = OpenAI(base_url='https://openrouter.ai/api/v1',
+                                 api_key=os.environ.get('OPENROUTER_API_KEY'),
+                                 timeout=httpx.Timeout(60.0, read=60.0, write=60.0, connect=60.0), max_retries=0)
+        return self.client
+
+    def classify(self, task_prompt, items_count, max_tokens_per_item=200):
+        client = self.get_client()
+        completion = client.chat.completions.create(
+            model=self.model_name,
+            messages=[
+                {"role": "user", "content": task_prompt}
+            ],
+            max_tokens=(max_tokens_per_item * items_count) + 15,
+            temperature=0.5
+        )
+
```

Comment:
Can we check here for some of the OpenRouter specific errors (like out of tokens) so the model can error instead of passing downstream?

Reply:
trying to better understand the scope, can you please explain a bit more which cases this would cover, and what the end goal would be?

Reply:
Currently timpani sees errors like 'not enough credits', but doesn't know what that means, so it is going to keep sending additional requests.  I think if classcat is unable to do any processing, it should be reporting that as an exception, and perhaps crashing/going offline to signal for help?

Reply:
I see. I agree that it is helpful for the consumer of this service to distinguish between when it is ok to retry upon errors vs when it's a matter of service downtime and it's not ok to retry. I'll try to incorporate this into the code!

Reply:
Let's defer this until the PR for consistent error/status output and error handling is in place (this is in separate work)